MonstieWash/Assets/Scripts/MonsterAi/MoodArea.cs

```diff
@@ -0,0 +1,93 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+public class MoodArea : MonoBehaviour
+{
+    //References
+    private MonsterBrain m_mb;
+    private PlayerHand m_ph;
+
+
+    [Header("Configurables")]
+    [Tooltip("Select the layer of the mood area")] [SerializeField] private LayerMask layerMask;
+    [Tooltip("Add new moods to be affected by this, assign the type of mood and by what value it is affected.")] [SerializeField] private List<moodEffect> moodEffects; //Stores which moods will be changed and by how much.
+    [Tooltip("How frequently the area will react to being touched. (Every 'x' seconds).")] [SerializeField] private float areaCooldown; //How frequently the area will react to being touched.
+    [Tooltip("Toggle on to make the area reduce effectiveness over frequent use.")] [SerializeField] private bool diminishingEffectiveness; //When repeatedly touched will reduce its effects momentarily.
+    [Tooltip("How fast its effectiveness diminishes if above bool toggled on.")] [SerializeField] private int diminishStrength; //How quickly the effectiveness diminishes.
+    [Tooltip("Produces debug text in console when toggled on.")] [SerializeField] private bool debug;
+
+    //Internal states
+    private float currentCooldown;
+    private float currentEffectiveness; //Current effectiveness of area.
+
+    [Serializable]
+    public struct moodEffect
+    {
+        public MoodType mt; //Which mood to target.
+        public float reactionStrength; //How much the mood will change by.
+    }
+
+    private void OnEnable()
+    {
+        //Assign to Input system.
+        InputManager.Inputs.OnActivate_Held += TestTouch;
+
+        m_mb = FindFirstObjectByType<MonsterBrain>();
+        m_ph = FindFirstObjectByType<PlayerHand>();
+        currentEffectiveness = 100;
+    }
+
+
+    private void Update()
+    {
+        //Restore Effectiveness over time.
+        if (diminishingEffectiveness) currentEffectiveness = Mathf.Clamp(currentEffectiveness +( (diminishStrength/2) * Time.deltaTime), 0 , 100);
+        //Reduce Cooldown over time.
+        currentCooldown = Mathf.Clamp(currentCooldown -= Time.deltaTime, 0, areaCooldown);
```

Comment:
Just for these it's important when working with floats to keep everything a float so you don't get rounding errors.
```suggestion
        if (diminishingEffectiveness) currentEffectiveness = Mathf.Clamp(currentEffectiveness + ((diminishStrength/2f) * Time.deltaTime), 0f , 100f);
        //Reduce Cooldown over time.
        currentCooldown = Mathf.Clamp(currentCooldown -= Time.deltaTime, 0f, areaCooldown);
```